broker/src/main/java/io/camunda/zeebe/broker/system/partitions/RaftPartitionHealth.java

```diff
@@ -7,50 +7,55 @@
  */
 package io.camunda.zeebe.broker.system.partitions;
 
-import io.atomix.raft.RaftFailureListener;
 import io.atomix.raft.partition.RaftPartition;
 import io.camunda.zeebe.util.health.FailureListener;
 import io.camunda.zeebe.util.health.HealthMonitorable;
 import io.camunda.zeebe.util.health.HealthStatus;
 import io.camunda.zeebe.util.sched.ActorControl;
-import java.util.concurrent.CompletableFuture;
+import java.util.HashSet;
+import java.util.Set;
 
-public class RaftPartitionHealth implements HealthMonitorable, RaftFailureListener {
+public class RaftPartitionHealth implements HealthMonitorable, FailureListener {
```

Comment:
When this class was added, atomix was a different project. So this class acted as a glue between zeebe and atomix. But now we are more flexible in adding new things to raft. So we can also move this logic in to raft. `RaftPartition` can implement `HealthMonitorable`, which delegates its logic to `RaftContext`. `RaftContext` can either implement `FailureListener` or delegate it to RaftPartitionHealth. This would also remove a lot of redundant code that implement methods in FailureListener. Wdyt?